bittensor/_neuron/text/neuron_utilities.py

```diff
@@ -0,0 +1,50 @@
+import bittensor
+import torch
+import torch.nn.functional as F
+
+
+def joining_context(return_ops, topk_weights, responses):
+    # ---- Join based on weights ----
+    joining_uids= torch.where( return_ops == bittensor.proto.ReturnCode.Success )[0]
+    joining_weights = F.softmax( topk_weights[(return_ops == bittensor.proto.ReturnCode.Success)], dim = 0 ) 
+    output = torch.zeros( (responses[0].shape[0], responses[0].shape[1], bittensor.__network_dim__))
+    for index, joining_weight in enumerate( joining_weights ):
+        output += responses[joining_uids[index]]* joining_weight
+    return output, joining_uids
+
+def jacobian(y, x, create_graph=False,hessian =False):                                                               
+    jac = []                                                                                          
+    flat_y = y.reshape(-1)                                                                            
+    grad_y = torch.zeros_like(flat_y)                                                                 
+    for i in range(len(flat_y)): 
+        if hessian ==True and flat_y[i].item() == 0:
+            grad_x = torch.zeros_like(x)
+            jac.append(grad_x.reshape(x.shape)) 
+            pass
+        else:
+            grad_y[i] = 1.
+            grad_x, = torch.autograd.grad(flat_y, x, grad_y, retain_graph=True, create_graph=create_graph)
+            jac.append(grad_x.reshape(x.shape))                                                           
+            grad_y[i] = 0.                                                                                
+    return torch.stack(jac).reshape(y.shape + x.shape)     
+
+def fisher_score_approixmation(loss, peer_weights, ):
```

Comment:
Nit: approximation, not approixmation, typo :)

Reply:
Also, is this function vestigial? I don't see it being used anywhere.

Reply:
Haha nice catch, yes, it is vestigial. I left it incase we wish to go back to fisher approximation in the future.